src/components/badge/badge.spec.tsx

```diff
@@ -54,4 +61,79 @@ describe("Badge", () => {
       expect(wrapper.find(BADGE).exists()).toBe(false);
     });
   });
+
+  describe("Accessibility", () => {
+    const ariaLabelText = "Generic aria message";
+    const wrapper = mount(
+      renderComponent({ counter: 9, "aria-label": ariaLabelText })
+    );
+
+    it("should have the relevant aria-label when aria-label is specified", () => {
+      expect(wrapper.find(Badge).first().prop("aria-label")).toBe(
+        ariaLabelText
+      );
+    });
+  });
+
+  describe("Styles", () => {
+    it("should apply the correct style when onClick is present", () => {
+      const wrapper = mount(renderComponent({ counter: 9, onClick: () => {} }));
+      assertStyleMatch(
+        {
+          padding: `0`,
```

Comment:
I don't think we should be checking styles here considering that we have it covered with chromatic. I'd just test that depending on wether we have the onClick present or not, we're rendeing either a span or a button

Reply:
If you think they aren't needed, I'll get them removed and a test written like you've said here 😄

Reply:
If you went with a polymorphic approach you could` find("button")` or `find("span")` and assert the correct element exists and that would suffice imo. Alternatively yo could test the cursor styling here as that won't be caught in chromatic, but the rest of the styling will be as @mkrds says